mtg_card_identifier/CardScanner/CardScannerView.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  CardScannerView.swift
+//  mtg_card_identifier
+//
+//  Created by mac on 8/16/23.
+//
+
+import SwiftUI
+import VisionKit
+import AVFoundation
+
+struct CardScannerView: View {
+    @State private var cameraAccessAuthorized = false
+
+    var body: some View {
+        NavigationView {
+            if cameraAccessAuthorized {
+                CardScanner()
+            } else {
+                NoCameraAccessView()
+            }
+        }
+        .onAppear {
+            requestCameraAccess()
+            checkCameraAuthorizationStatus()
```

Comment:
I’d pull this out into the init.

Reply:
So like this?   init() {
            requestCameraAccess()
            checkCameraAuthorizationStatus()
        }